packages/fxa-shared/metrics/amplitude-event.1.schema.json

```diff
@@ -221,12 +254,115 @@
         }
       },
       "then": {
-        "required": ["language"]
+        "required": ["language"],
+        "properties": {
+          "user_properties": {
+            "required": ["flow_id"]
+          },
+          "event_properties": {
+            "required": ["plan_id", "product_id"]
+          }
+        }
+      }
+    },
+    {
+      "if": {
+        "$comment": "Additional required fields for fxa_pay_setup* events",
+        "required": ["event_type"],
+        "properties": {
+          "event_type": {
+            "type": "string",
+            "pattern": "^fxa_pay_setup -"
+          }
+        }
+      },
+      "then": {
+        "properties": {
+          "event_properties": {
+            "required": ["checkout_type"]
+          }
+        }
+      }
+    },
+    {
+      "if": {
+        "$comment": "Additional required fields for fxa_pay_subscription_change* events",
+        "required": ["event_type"],
+        "properties": {
+          "event_type": {
+            "type": "string",
+            "pattern": "^fxa_pay_subscription_change -"
+          }
+        }
+      },
+      "then": {
+        "properties": {
+          "event_properties": {
+            "required": ["previous_plan_id", "previous_product_id", "subscription_id", "payment_provider"]
```

Comment:
Blocking change request:

I'm seeing validation errors locally for the `fxa_pay_subscription_change - view` and ` - engage` events saying `previous_plan_id`, `previous_product_id` and `payment_provider` are missing. Can you update these two events to include these missing required properties?

I do see them on the corresponding `- submit` and `- success` events, so hopefully it's not too difficult.